### ISSUE TITLE
Parallel visualization with GLVis

### ISSUE THREAD

**Issue by hughcars** [mfem/mfem#3563]
I'm currently attempting to visualize `ex15p`. Running `ex15`, GLVis shows the circle adaptation front propagating through the star domain just fine. Running ex15p, with `mpirun -np 1`, I receive an error "GLVis quit unexpectedly" (this is on a mac M1). This is with the vended macOS image from https://glvis.org/download/. Looking in the resulting error report, it has `mfem::Mesh::ReadMFEMMesh(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, int&)` throwing an error.

Building GLVis from source in Debug mode and running the same, I can then get a more full error message:
```
new connection: parallel 1 0
connection[0]: reading initial data ...  type solution ... 

Verification failed: (ident == "boundary") is false:
 --> invalid mesh file
 ... in function: void mfem::Mesh::ReadMFEMMesh(std::istream &, int, int &)
 ... in file: [full path redacted]/mfem/mesh/mesh_readers.cpp:68

Backtrace:
0) [0xc16000104611187]: mfem::mfem_error(char const*)
1) [0xbc47000104778b9b]: mfem::Mesh::ReadMFEMMesh(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, int&)
2) [0x401e800104721f03]: mfem::Mesh::Loader(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)
3) [0xca7d00010471d40b]: mfem::Mesh::Load(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, int, bool)
4) [0x892c000104720287]: mfem::Mesh::Mesh(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, int, bool)
5) [0x1568800104720537]: mfem::Mesh::Mesh(std::__1::basic_istream<char, std::__1::char_traits<char> >&, int, int, bool)
6) [0xb05100010457ae3f]: StreamState::ReadStreams(std::__1::vector<std::__1::unique_ptr<std::__1::basic_istream<char, std::__1::char_traits<char> >, std::__1::default_delete<std::__1::basic_istream<char, std::__1::char_traits<char> > > >, std::__1::allocator<std::__1::unique_ptr<std::__1::basic_istream<char, std::__1::char_traits<char> >, std::__1::default_delete<std::__1::basic_istream<char, std::__1::char_traits<char> > > > > > const&)
7) [0xdf768001044d5267]: GLVisServer(int, bool, bool, bool)
8) [0x4b098001044f0c3f]: decltype(std::declval<void (*)(int, bool, bool, bool)>()(std::declval<int>(), std::declval<bool>(), std::declval<bool>(), std::declval<bool>())) std::__1::__invoke[abi:v160000]<void (*)(int, bool, bool, bool), int, bool, bool, bool>(void (*&&)(int, bool, bool, bool), int&&, bool&&, bool&&, bool&&)
9) [0xed050001044f0bdb]: void std::__1::__thread_execute[abi:v160000]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(int, bool, bool, bool), int, bool, bool, bool, 2ul, 3ul, 4ul, 5ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(int, bool, bool, bool), int, bool, bool, bool>&, std::__1::__tuple_indices<2ul, 3ul, 4ul, 5ul>)
10) [0xcb4c0001044f0837]: void* std::__1::__thread_proxy[abi:v160000]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(int, bool, bool, bool), int, bool, bool, bool> >(void*)
11) [0xa31e80019d5de06b]: _pthread_start

Lookup backtrace source lines:
atos -o [full path redacted]/glvis/build/glvis -l 0x1044c8000 0xc16000104611187 0xbc47000104778b9b 0x401e800104721f03 0xca7d00010471d40b 0x892c000104720287 0x1568800104720537 0xb05100010457ae3f 0xdf768001044d5267 0x4b098001044f0c3f 0xed050001044f0bdb 0xcb4c0001044f0837
atos -o /usr/lib/system/libsystem_pthread.dylib -l 0x19d5d7000 0xa31e80019d5de06b

zsh: abort      ./glvis
```
Any ideas on what I am doing wrong here?

**Comment by v-dobrev** [comment]
I cannot reproduce this failure on Mac with Apple silicon when using the latest `master` branches of both MFEM and GLVis.

@hughcars, were you able to resolve the problem?

**Comment by hughcars** [comment]
Hi @v-dobrev yes, it turned out to be something unrelated so I closed, not sure what was happening but I git bisected and removed the offender. I had set [`enum Type`](https://github.com/mfem/mfem/blob/master/fem/geom.hpp#L35) to `enum Type : signed char`, as there's a few places through NCMesh etc. where there's casting back and forth between `char`, `signed char` and `int`, ostensibly to save memory, that I was trying to understand, and it only has `-1:7` values. Undoing this resolved things, why it failed in the first place remains a mystery, as the symptom was that [this line](https://github.com/mfem/mfem/blob/master/mesh/mesh_readers.cpp#L66) never loaded anything into ident, i.e. a while loop placed around checking that it changed from `"elements"` ran forever. Some kind of overflow error, but I decided it wasn't worth hunting down.
